runtime/lua/vim/filetype.lua

```diff
@@ -863,6 +898,184 @@ local extension = {
       return "text"
     end
   end,
+  cmd = function (path, bufnr)
+    if getline(bufnr, 1):find("^/%*") then
+      return "rexx"
+    end
+    return "dosbatch"
+  end,
+  rul = function(path, bufnr)
+    if getline(bufnr, 1, 6):lower():find("installshield") then
+      return "ishd"
+    end
+    return "diva"
+  end,
+  cpy = function(path, bufnr)
+    if getline(bufnr, 1):find("^##") then
+      return "python"
+    end
+    return "cobol"
+  end,
+  dsl = function(path, bufnr)
+    if getline(bufnr, 1):find("^%s*<!") then
+      return "dsl"
+    end
+    return "structurizr"
+  end,
+  edf = "edif",
+  edfi = "edif",
+  edo = "edif",
+  edn = function(path, bufnr)
+    if getline(bufnr, 1):find("^%s*%(%s*edif>") then
+      return "edif"
+    end
+    return "clojure"
+  end,
+  smil = function(path, bufnr)
+    if getline(bufnr, 1):find("<?%s*xml.*?>") then
+      return "xml"
+    end
+    return "smil"
+  end,
+  smi = function(path, bufnr)
+    if getline(bufnr, 1):find("<smil>") then
+      return "smil"
+    end
+    return "mib"
+  end,
+  install = function(path, bufnr)
+    if getline(bufnr, 1):find("<%?php") then
+      return "php"
+    end
+    vim.fn["dist#ft#SetFileTypeSH"]("bash")
+  end,
+  pm = function(path, bufnr)
+    if getline(bufnr, 1):find("XPM2") then
+      return "xpm2"
+    elseif getline(bufnr, 1):find("XPM") then
+      return "xpm"
+    else
+      return "perl"
+    end
+  end,
+  me = function(path, bufnr)
+    local filename = vim.fn.fnamemodify(path, ":t")
+    if filename ~= "read.me" and filename ~= "click.me" then
+      return "nroff"
+    end
+  end,
+  reg = function(path, bufnr)
+    local line = getline(bufnr, 1)
```

Comment:
Should be changed to `local line = getline(bufnr, 1):lower()` (because Vim uses `=~?` here) while also changing all upper case letters in the regex to lowercase (e.g. `REGEDIT` => `regedit`).